utils/handlers.go

```diff
@@ -0,0 +1,44 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package utils
+
+import (
+	"context"
+	"log"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func HandleMissingWatchers(c client.Client, crdName []string) error {
+	var crdsPresent int
+	for i := range crdName {
+		crd := apiextensionsv1.CustomResourceDefinition{}
+		crd.Name = crdName[i]
+		key := client.ObjectKeyFromObject(&crd)
+		if err := c.Get(context.Background(), key, &crd); err != nil {
+			return err
+		} else {
+			crdsPresent++
+			log.Printf("agent restart required to register watchers for the CRD %s", crdName[i])
+		}
+	}
+	if crdsPresent == len(crdName) {
+		log.Fatal("all missing CRDs present on cluster, restarting the agent to register watchers")
+	}
+	return nil
```

Comment:
Why does "handle" utils do "get"s?
I would expect it to just be a log operation followed by a fatal operation

Also, handlers as a filename have no context. Please move to the appropriate util file.

Reply:
renamed the file to `restart.go` this can be renamed or moved to a different file later in the future when there is a suitable util file